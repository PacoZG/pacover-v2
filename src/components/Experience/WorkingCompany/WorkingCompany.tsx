import React from 'react'
import { useTranslations } from 'next-intl'
import '../experience-style.css'

interface Experience {
  company: string
  period: string
  position: string
  responsabilities: string
}

interface WorkingCompanyProps {
  experience: Experience
  link?: string
}

const WorkingCompany: React.FC<WorkingCompanyProps> = ({ experience, link }) => {
  const t = useTranslations('experience')

  const getProcessedResponsibilities = (responsibilitiesContent: string) => {
    if (typeof responsibilitiesContent === 'string') {

      const lines: string[] = responsibilitiesContent.split('\n').filter(line => line.trim() !== '')

      return (
        <ul className="list-disc list-inside space-y-1">
          {lines.map((item, index) => (
            <li key={index} className="text-gray-700 dark:text-gray-400">
              {item.trim()}
            </li>
          ))}
        </ul>
      )
    }

    return <p className="text-gray-700 dark:text-gray-400">{responsibilitiesContent}</p>
  }

  return (
    <div key={experience.company} className="experience-card">
      <div className="flex flex-col">
        <div className="flex flex-col">
          <h1 className="experience-card-title">
            {link && (
              <a
                experience-tool-tip-msg={t('tool-tip-msg')}
                className="experience-card-link"
                href={link}
                target="_blank"
                rel="noopener noreferrer"
              >
                {experience.company}
              </a>
            )}
            {!link && experience.company}
          </h1>

          <div className="experience-card-period">{experience.period}</div>
        </div>

        <div className="experience-card-position">{experience.position}</div>
      </div>

      <div className="experience-card-responsibilities">
        {getProcessedResponsibilities(experience.responsabilities)}
      </div>
    </div>
  )
}

export default WorkingCompany
