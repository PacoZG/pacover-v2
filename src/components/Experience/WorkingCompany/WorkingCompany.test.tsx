import React from 'react'
import { render } from '@testing-library/react'
import { describe, test, expect, vi, beforeEach } from 'vitest'
import WorkingCompany from '@/components/Experience/WorkingCompany/WorkingCompany'

describe('Experience', () => {
  describe('given configured with dependencies', () => {
    beforeEach(() => {
      window.scrollTo = vi.fn()

      vi.mock('next-intl', () => ({
        useTranslations: () => (key: string) => key,
      }))
    })

    describe('WorkingCompany component', () => {
      test('matches the snapshot', () => {
        const { container } = render(
          <WorkingCompany
            experience={{
              company: 'some-company',
              period: 'some-period',
              position: 'some-position',
              responsabilities: 'some-resp',
            }}
          />
        )

        expect(container).toMatchSnapshot()
      })
    })
  })
})
