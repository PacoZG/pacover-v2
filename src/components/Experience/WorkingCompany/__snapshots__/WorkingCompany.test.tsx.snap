// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Experience > given configured with dependencies > WorkingCompany component > matches the snapshot 1`] = `
<div>
  <div
    class="experience-card"
  >
    <div
      class="flex flex-col"
    >
      <div
        class="flex flex-col"
      >
        <h1
          class="experience-card-title"
        >
          some-company
        </h1>
        <div
          class="experience-card-period"
        >
          some-period
        </div>
      </div>
      <div
        class="experience-card-position"
      >
        some-position
      </div>
    </div>
    <div
      class="experience-card-responsibilities"
    >
      <ul
        class="list-disc list-inside space-y-1"
      >
        <li
          class="text-gray-700 dark:text-gray-400"
        >
          some-resp
        </li>
      </ul>
    </div>
  </div>
</div>
`;
